serde-generate/src/solidity.rs

```diff
@@ -1045,13 +1045,19 @@ function bcs_serialize_{name}({name} input)
     pure
     returns (bytes memory)
 {{
+    uint8 value0;
+    uint8 value1;
     if (input == {name}.None) {{
-        return abi.encodePacked(0);
+        value0 = 0;
+        return abi.encodePacked(value0);
```

Comment:
Why the intermediate variables?

Reply:
Because the `abi.encodePacked(0)` does not pass the compilation stage.